### PR TITLE
[AST builder] Implement Insert statement

### LINKED ISSUE #712: [AST Builder] Implement Insert Statement
## Description
Implement `insert` statement ast builder

## Background
refer issue #614 

## Task
``` sql
        INSERT INTO Test (id, num, name) VALUES
            (1,     2,     "Hello"),
            (-(-1), 9,     "World"),
            (+3,    2 * 2, "Great");

        INSERT INTO fizzbuzz (id, num, name) VALUES
            (1,     1,     "fizz"),
            (-(-2), 2,     "buzz"),
            (+3,    3 * 3, "fizzbuzz");
```

## Task to Test
```rs
   fn insert(){
        let actual = table("Foo")
                    .insert()
                    .values(vec![
                        vec![1, 5],
                        vec![2, 3],
                    ])
                    .build();
        let expected = r#"INSERT INTO Foo VALUES (1, 5), (2, 3)"#;
        test(actual, expected);

        let actual = table("Foo")
                    .insert()
                    .columns("id, name")
                    .values(vec![
                        vec![1, 5],
                        vec![2, 3],
                    ])
                    .build();
        let expected = r#"INSERT INTO Foo (id, name) VALUES (1, 5), (2, 3))"#;
        test(actual, expected);

        let actual = table("Foo")
                    .insert()
                    .columns(vec!["hi"])
                    .values(vec![vec![num(7)]])
                    .build();
        let expected = r#"INSERT INTO Foo (hi) VALUES (7)"#;
        test(actual, expected);

        let actual = table("Foo")
                    .insert()
                    .as_select(table("Bar").select("id, name").limit(10).project("id, name"))
                    .build();
        let expected = r#"INSERT INTO Foo SELECT id, name FROM Bar LIMIT 10"#;
        test(actual, expected);
    }
```

### PR DESCRIPTION
**Pull request by kim-seo-hyun**
## Background
resolve #712

### CODE REVIEW

File: core/src/ast_builder/insert.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+#[cfg(test)]
+mod tests {
+    use crate::ast_builder::{num, table, test};
+
+    #[test]
+    fn insert() {
+        let actual = table("Foo")
+            .insert()
+            .values(vec![vec![num(1), num(5)], vec![num(2), num(3)]])
```
Comment (panarch) [inline core/src/ast_builder/insert.rs:74]:
```rust
.values(vec!["1, 5", "2, 3"])
```

File: core/src/ast_builder/query.rs [hunk 2]
```diff
@@ -83,9 +91,26 @@ impl TryFrom<QueryNode> for Query {
             QueryNode::LimitOffset(query_node) => query_node.prebuild().map(NodeData::build_query),
             QueryNode::Offset(query_node) => query_node.prebuild().map(NodeData::build_query),
             QueryNode::OffsetLimit(query_node) => query_node.prebuild().map(NodeData::build_query),
+            QueryNode::Project(query_node) => query_node.prebuild().map(NodeData::build_query),
             QueryNode::Text(query_node) => {
                 parse_query(query_node).and_then(|item| translate_query(&item))
             }
+            QueryNode::Values(values) => {
+                // let values: Vec<ExprList> = values;
```
Comment (24seconds) [inline core/src/ast_builder/query.rs:99]: [request]
remove this comments

**Review by 24seconds — COMMENTED** [review]
Looks good overall

File: core/src/ast_builder/query.rs [hunk 3]
```diff
@@ -83,9 +91,26 @@ impl TryFrom<QueryNode> for Query {
...
+                let values: Vec<Vec<Expr>> = values
+                    .into_iter()
+                    // .map(|expr_list: ExprList| expr_list.try_into()) // Vec<Expr>
```
Comment (24seconds) [inline core/src/ast_builder/query.rs]: [ditto]
Remove comments

**Review by 24seconds — APPROVED** [review]
LGTM!

**Review by panarch — APPROVED** [review]
Looks all nice! Now our AST builder supports insert statement!
Thanks a lot 👍 👍

**Pull request merged**